test/application/ladders-matchmaker.js

```diff
@@ -109,14 +116,13 @@ describe('Matchmaker', function () {
 		assert.strictEqual(matchmaker.searches.get(formatid).size, 0);
 	});
 
-	// FIXME: a race condition in battles and sockets breaks this test
-	it.skip('should create a new battle room after matchmaking', function () {
+	it('should create a new battle room after matchmaking', function () {
 		let formatid = 'gen7ou';
 		let {startBattle} = matchmaker;
 		matchmaker.startBattle = (...args) => {
 			matchmaker.startBattle = startBattle;
-			let room = matchmaker.startBattle(...args);
-			assert.ok(room instanceof Rooms.BattleRoom);
+			this.room = matchmaker.startBattle(...args);
```

Comment:
The room can't be destroyed outside of `afterEach`'s callback without making the test asynchronous. `this.room` may end up showing up again if bugs related to `Matchmaker#startBattle` pop up to merit more tests